src/Microsoft.Azure.AppConfiguration.AspNetCore/AzureAppConfigurationRefreshExtensions.cs

```diff
@@ -23,14 +24,21 @@ public static IApplicationBuilder UseAzureAppConfiguration(this IApplicationBuil
                 throw new ArgumentNullException(nameof(builder));
             }
 
+            IConfigurationRefresherProvider refresherProvider = (IConfigurationRefresherProvider)builder.ApplicationServices.GetService(typeof(IConfigurationRefresherProvider));
+
             // Verify if AddAzureAppConfiguration was done before calling UseAzureAppConfiguration.
             // We use the IConfigurationRefresherProvider to make sure if the required services were added.
-            if (builder.ApplicationServices.GetService(typeof(IConfigurationRefresherProvider)) == null)
+            if (refresherProvider == null)
+            {
+                throw new InvalidOperationException("Unable to find the required services. Please add all the required services by calling 'IServiceCollection.AddAzureAppConfiguration' in the application startup code.");
+            }
+
+            if (refresherProvider.Refreshers?.Count() != 0)
```

Comment:
If Refreshers is null, this condition will still be true (null != 0).
You could use `if (refresherProvider.Refreshers?.Count() > 0)`